src/ToggleSaveContext.js

```diff
@@ -0,0 +1,22 @@
+import React, { createContext, useContext, useState } from 'react';
+
+
+const ToggleSaveContext = createContext();
+
+export const useToggleSave = () => {
+  return useContext(ToggleSaveContext);
+};
+
+export const ToggleSaveProvider = ({ children }) => {
+  const [isToggled, setIsToggled] = useState(false);
+
+  const toggle = () => {
+    setIsToggled(!isToggled);
+  };
+
+  return (
+    <ToggleSaveContext.Provider value={{ isToggled, toggle }}>
+      {children}
+    </ToggleSaveContext.Provider>
+  );
```

Comment:
Never new these method's existed and very cool to learn about them. Awesome work!

Reply:
Context.API

https://www.freecodecamp.org/news/context-api-in-react/